years/2020/01/solutions.ts

```diff
@@ -0,0 +1,33 @@
+export const calculate = (input: number[], num: number) => {
+  const match = combinations(input, num).find((combination: number[]) => {
+    return combination.reduce((calculation: number, currentNumber: number) => calculation + currentNumber, 0) === 2020
+  })
+
+  if (!match) {
+    throw new Error(`Sorry, couldn't find a match`)
+  }
+
+  return match.reduce((a: number, b: number) => a * b, 1)
+}
+
+function combinations(set: number[], k: number): number[][] {
+  if (k > set.length) return []
+  if (k == set.length) return [set]
+
+  if (k == 1) {
+    let con: number[][] = []
+    set.forEach((_, index: number) => {
+      con.push([set[index]])
+    })
```

Comment:
Few things here:

1. If we're simply accessing each item by index, we might as well use the first arg in `forEach` instead:

```suggestion
    set.forEach(item => {
      con.push([item])
    })
```

2. I believe `forEach` should already be typed, so I think you can remove this

```suggestion
    set.forEach((_, index) => {
      con.push([set[index]])
    })
```

3. I think all we're doing here is iterating through each item in `set` and wrapping it in an array, so a much simpler way would be to use `.map`

```suggestion
    return set.map(item) => [item])
```

..then I think you can remove `let con: number[][] = []` and `return con`

I also have no idea what `con` is/means - perhaps a more descriptive name would help